packages/dev/core/src/Meshes/Compression/dracoCompression.ts

```diff
@@ -433,16 +232,16 @@ export class DracoCompression implements IDisposable {
      * Returns a promise that resolves when ready. Call this manually to ensure draco compression is ready before use.
      * @returns a promise that resolves when ready
      */
-    public whenReadyAsync(): Promise<void> {
+    public async whenReadyAsync(): Promise<void> {
         if (this._workerPoolPromise) {
-            return this._workerPoolPromise.then(() => {});
+            await this._workerPoolPromise;

```

Comment:
For my personal culture: is there a reason to do this instead of sending back a `Promise` directly as in the old code?

Reply:
This is a lot about readability.

The code before (.then(() => {});) was added only because of typescript - it takes a promise and generates another promise that returns nothing.
What I changed is make the function async, guarantying that a promise is returned. So if you just await a promise and don't return it, a Promise<void> is returned

As a dev calling this function, nothing changes. If you don't await the function call, you won't be blocked. 

I don't mind reverting it! I just find the extra promise confusing.